Change to get temp from HmIP-STHO-A Sensor

a-wolter opened the pull request:
ACTUAL_TEMPERATURE is needed to fetch the temperature from the HmIP-STHO-A sensor, which was added to pyhomematic https://github.com/danielperna84/pyhomematic/pull/125

## Description:

ACTUAL_TEMPERATURE was ignored, but it required to get the temp. from the STHO-A Sensor. This change implements an exception list.

**Related issue (if applicable):** fixes #13014 

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) 



```

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [ ] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [ ] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

a-wolter committed:
ACTUAL_TEMPERATURE is needed to fetch the temperature from the HmIP-STHO-A sensor.

homeassistant commented:
Hi @a-wolter, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#13048).

Once you do that we will be able to review and accept this pull request.

Thanks!

a-wolter committed:
fix typo

danielperna84 commented:
It may make sense to refactor the `HM_IGNORE_DISCOVERY_NODE` list to somehow include exceptions. In most cases ignoring the node makes sense. We do know about the [class_name](https://github.com/home-assistant/home-assistant/blob/e910ecfd5f642c80161c0bf243c06e29fd6a1913/homeassistant/components/homematic/__init__.py#L491). So if we would use a dict with the `class_name` as the key and a list with exceptions as the values, we would have more control. There probably will be more IP devices coming were we will get the same conflicting device-parameters.

Example of how the new dict could look like:
```
HM_IGNORE_DISCOVERY_NODE = {
    'ACTUAL_TEMPERATURE': ['IPAreaThermostat'],
    'ACTUAL_HUMIDITY': []
}
```

a-wolter committed:
Added new pyhomematic class to sensor list

a-wolter committed:
- Added exception list to exclude some classes from beeing ignored.
- Restored ACTUAL_TEMPERATURE in HM_IGNORE_DISCOVERY_NODE
- Changed required version for pyhomematic, since IPAreaThermostat is included in version 0.1.40

a-wolter committed:
fix for pylint / flake8 checks

a-wolter committed:
Fix indention

a-wolter committed:
requirements

a-wolter committed:
requirement file deleted

danielperna84 commented:
Please remove the changes of the requirements and the version. I will create another PR to raise the version as soon as the new pyhomematic version is released. You code technically works independent of the used pyhomematic version, so there is no need to reference a not yet existing version.

a-wolter committed:
Set required Version of pyhomematic back to 0.1.39

a-wolter committed:
removed empty line
